src/operator/subgraph/static_shape_subgraph_property.cc

```diff
@@ -79,26 +79,35 @@ class StaticShapeSubgraphProperty: public SubgraphProperty {
 
   void PrePartition(const nnvm::Graph& g,
                     const std::unordered_map<std::string, std::string>& options_map) override {
-    // update static_alloc and static_shape flags
-    if (options_map_.size() == 0) {
-      for (auto& kv : options_map) {
-        options_map_.emplace_back(kv);
-      }
-    }
-    if (param_name_set_.size() > 0) {
-      param_name_set_.clear();
-    }
-    // generate param_name_set_ for data_indices and param_indices
-    if (g.HasAttr("param_indices_list")) {
-      const std::vector<int>& param_indices_list
-          = g.GetAttr<std::vector<int>>("param_indices_list");
-      const auto& indexed_graph = g.indexed_graph();
-      for (const auto index : param_indices_list) {
-          auto nid = indexed_graph.input_nodes()[index];
-          nnvm::Node n = *(indexed_graph[nid].source);
-          param_name_set_.emplace(n.attrs.name);
+    options_map_.clear();
+    param_name_set_.clear();
+    const auto& indexed_graph = g.indexed_graph();
+    for (auto& kv : options_map) {
+      // update static_alloc and static_shape flags
+      if (kv.first == "static_alloc" || kv.first == "static_shape") {
+        if (kv.second == "True") {
+          options_map_.emplace_back(kv.first, "true");
+        } else {
+          options_map_.emplace_back(kv.first, "false");
+        }
```

Comment:
@waytrue17 can this be simplified to `options_map_.emplace_back(kv.first, kv.second);`?